### PR TITLE
Add assembly Carthage artifacts

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [ ] ~Has `CHANGELOG.md` been updated?~
* [ ] ~Are tests passing locally?~
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests?~
* [x] Did you test your change with the sample apps?

## Description

Add assembly Carthage artifacts.

## Related PRs or issues

[AB#86950](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/86950)

### CODE REVIEW

File: Scripts/create-archive.sh [hunk 1]
```diff
@@ -7,18 +7,20 @@ set -e
 PROJECT_DIR="$(dirname "$0")/.."
 
 # Create temporary directory
-TEMP_DIR=$(mktemp -d -t "$1") 
-mkdir -p "$TEMP_DIR/$1"
+TEMP_ARCH_NAME="PLCrashReporter"
```
Comment (MatkovIvan) [inline Scripts/create-archive.sh:10]: You can use project/target name instead of hardcoding it

File: Scripts/create-archive.sh [hunk 2]
```diff
@@ -7,18 +7,20 @@ set -e
...
+TEMP_PATH_TO_ARCH="$TEMP_DIR/$TEMP_ARCH_NAME"
+mkdir -p "$TEMP_PATH_TO_ARCH"
 
 # Copy required files
-cp "$PROJECT_DIR/LICENSE" "$TEMP_DIR/$1/LICENSE.txt"
-cp -R "../Documentation" "$TEMP_DIR/$1"
-(cd "$TEMP_DIR/$1" && ln -sf "Documentation/index.html" "API Documentation.html")
-cp -R "${@:2}" "$TEMP_DIR/$1"
+cp "$PROJECT_DIR/LICENSE" "$TEMP_PATH_TO_ARCH/LICENSE.txt"
+#cp -R "../Documentation" "$TEMP_PATH_TO_ARCH"
```
Comment (MatkovIvan) [inline Scripts/create-archive.sh:17]: No need to keep commented code in the repo

File: Scripts/create-archive.sh [hunk 3]
```diff
@@ -7,18 +7,20 @@ set -e
 PROJECT_DIR="$(dirname "$0")/.."
 
 # Create temporary directory
-TEMP_DIR=$(mktemp -d -t "$1") 
-mkdir -p "$TEMP_DIR/$1"
+TEMP_ARCH_NAME=`echo $1 | cut -d'-' -f 1`
```
Comment (DmitriyKirakosyan) [inline Scripts/create-archive.sh:10]: Why do we need "| cut -d'-' -f 1`" here. It wasn't there before the changes.

File: Scripts/create-archive.sh [hunk 4]
```diff
@@ -7,18 +7,20 @@ set -e
...
+cp "$PROJECT_DIR/LICENSE" "$TEMP_PATH_TO_ARCH/LICENSE.txt"
+cp -R "../Documentation" "$TEMP_PATH_TO_ARCH"
+(cd "$TEMP_PATH_TO_ARCH" && ln -sf "Documentation/index.html" "API Documentation.html")
+cp -R "${@:2}" "$TEMP_PATH_TO_ARCH"
 
 # Archive content
-rm -f "$TEMP_DIR/$1.zip"
-(cd "$TEMP_DIR" && zip -ryq9 "$1.zip" "$1")
+rm -f "$TEMP_DIR/$TEMP_ARCH_NAME.zip"
+(cd "$TEMP_DIR" && zip -ryq9 "$1.zip" "$TEMP_ARCH_NAME")
```
Comment (DmitriyKirakosyan) [inline Scripts/create-archive.sh:23]:
```suggestion
(cd "$TEMP_DIR" && zip -ryq9 "$TEMP_ARCH_NAME.zip" "$TEMP_ARCH_NAME")
```

File: Scripts/create-archive.sh [hunk 5]
```diff
@@ -7,18 +7,20 @@ set -e
...
 mv "$TEMP_DIR/$1.zip" .
```
Comment (DmitriyKirakosyan) [inline Scripts/create-archive.sh:24]:
```suggestion
mv "$TEMP_DIR/$TEMP_ARCH_NAME.zip" .
```

File: Scripts/create-archive.sh (hunk 3 above)
Comment (AnastasiaKubova) [reply Scripts/create-archive.sh:10]: to avoid hardcoding of file name now the name will be got from result-name parameter without version

Comment (DmitriyKirakosyan) [inline Scripts/create-archive.sh:10]: why is it named with prefix "TEMP_"?

Comment (DmitriyKirakosyan) [reply Scripts/create-archive.sh:10]: Let's use $PRODUCT_NAME and fallback to "| cut -d'-' -f 1`" if it is not defined?

**Comment by DmitriyKirakosyan** [comment]
@AnastasiaKubova  The changes will be applied to all binaries. Have you tested this with cocoapods?

File: .azure-pipelines/azure-pipelines.yml [hunk 6]
```diff
@@ -106,6 +106,7 @@ jobs:
       VERSION="$(cd $BUILD_SOURCESDIRECTORY && agvtool vers -terse)"
       [[ $BUILD_SOURCEBRANCH != 'refs/heads/master' ]] && VERSION="$VERSION+$(cd $BUILD_SOURCESDIRECTORY && git rev-parse --short $BUILD_SOURCEVERSION)"
       "$BUILD_SOURCESDIRECTORY/Scripts/create-archive.sh" "PLCrashReporter-$VERSION" "iOS Framework" "tvOS Framework" "Mac OS X Framework" "Tools"
+      "$BUILD_SOURCESDIRECTORY/Scripts/create-archive.sh" "PLCrashReporter-$VERSION.carthage.framework" "iOS Framework" "tvOS Framework" "Mac OS X Framework" "Tools"
```
Comment (DmitriyKirakosyan) [inline .azure-pipelines/azure-pipelines.yml:109]: Should we add this in Xcode as well?

Screenshot 2021-06-09 at 11 39 43

**Review by DmitriyKirakosyan — APPROVED** [review]

**Pull request merged**